prov/mrail/src/mrail_attr.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018 Intel Corporation, Inc.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "mrail.h"
+
+struct fi_tx_attr mrail_tx_attr = {
+	.caps 		= ~0x0ULL,
+	.msg_order 	= ~0x0ULL,
+	.comp_order 	= ~0x0ULL,
+	.inject_size 	= SIZE_MAX,
+	.size 		= SIZE_MAX,
+	.iov_limit 	= SIZE_MAX,
+	.rma_iov_limit 	= SIZE_MAX,
+};
+
+struct fi_rx_attr mrail_rx_attr = {
+	.caps 			= ~0x0ULL,
+	.msg_order 		= ~0x0ULL,
+	.comp_order 		= ~0x0ULL,
+	.total_buffered_recv 	= SIZE_MAX,
+	.size 			= SIZE_MAX,
+	.iov_limit		= SIZE_MAX,
+};
+
+struct fi_ep_attr mrail_ep_attr = {
+	.type 			= FI_EP_UNSPEC,
+	.protocol 		= FI_PROTO_MRAIL,
+	.protocol_version 	= 1,
+	.max_msg_size 		= SIZE_MAX,
+	.msg_prefix_size	= SIZE_MAX,
+	.max_order_raw_size 	= SIZE_MAX,
+	.max_order_war_size 	= SIZE_MAX,
+	.max_order_waw_size 	= SIZE_MAX,
+	.tx_ctx_cnt 		= SIZE_MAX,
+	.rx_ctx_cnt 		= SIZE_MAX,
+	.auth_key_size		= SIZE_MAX,
+};
+
+struct fi_domain_attr mrail_domain_attr = {
+	.name			= "ofi_mrail_domain",
+	.threading 		= FI_THREAD_SAFE,
+	.control_progress 	= FI_PROGRESS_AUTO,
+	.data_progress 		= FI_PROGRESS_AUTO,
```

Comment:
Probably need to set these to manual

Reply:
At least for the initial un-ordered messages implementation, the progress would be determined by the underlying provider. So setting this to AUTO ensures we don't reject an app's request before checking underlying provider.